server/service/osquery_utils/queries.go

```diff
@@ -1218,17 +1218,25 @@ func GetDetailQueries(fleetConfig config.FleetConfig, features *fleet.Features)
 		generatedMap["scheduled_query_stats"] = scheduledQueryStats
 	}
 
-	for _, env := range os.Environ() {
-		prefix := "FLEET_DANGEROUS_REPLACE_"
-		if !strings.HasPrefix(env, prefix) {
-			continue
+	if features != nil {
+		var unknownQueries []string
+
+		for name, override := range features.DetailQueryOverrides {
+			query, ok := generatedMap[name]
+			if !ok {
+				unknownQueries = append(unknownQueries, name)
+				continue
+			}
+			if override == nil {
+				delete(generatedMap, name)
+			} else {
+				query.Query = *override
+				generatedMap[name] = query
+			}
 		}
-		if i := strings.Index(env, "="); i >= 0 {
-			queryName := strings.ToLower(strings.TrimPrefix(env[:i], prefix))
-			newQuery := env[i+1:]
-			query := generatedMap[queryName]
-			query.Query = newQuery
-			generatedMap[queryName] = query
+
+		if len(unknownQueries) > 0 {
+			logging.WithErr(ctx, ctxerr.New(ctx, fmt.Sprintf("unknown-queries: %s", strings.Join(unknownQueries, ","))))
```

Comment:
a nit: but isn't this message a bit cryptic? how do I know that the unknown-queries are queries I'm trying to override?

Reply:
Added the feature name to it. Let me know if that helps. [60d3e72](https://github.com/fleetdm/fleet/pull/9117/commits/60d3e72d9c4ee0c4d7ed4a58f713bba420067ac4).